src/file.cpp

```diff
@@ -354,6 +347,47 @@ namespace libtorrent
 		// convert to seconds
 		return time_t(ft / 10000000 - posix_time_offset);
 	}
+
+	using win_path_string =
+#if TORRENT_USE_WSTRING
+		std::wstring;
+#else
+		std::string;
+#endif
+
+	void replace_separators(std::string &path)
+	{
+		for (auto& p : path)
+			if (p == '/')
+				p = '\\';
+	}
+
+	win_path_string convert_to_win_path_string(std::string const& path)
+	{
+#if TORRENT_USE_UNC_PATHS
+		std::string prepared_path;
+		// UNC paths must be absolute
+		// network paths are already UNC paths
+		if (path.substr(0,2) == "\\\\")
+			prepared_path = path;
+		else
+		{
+			std::string sep_path { path };
+			replace_separators(sep_path);
```

Comment:
instead of this function, please use: ``std::replace(sep_path.begin(), sep_path.end(), '/', '\\');``. That would work, right?